src/ngx_rewrite_driver_factory.cc

```diff
@@ -151,6 +151,10 @@ RewriteOptions* NgxRewriteDriverFactory::NewRewriteOptions() {
   return options;
 }
 
+RewriteOptions* NgxRewriteDriverFactory::NewRewriteOptionsForQuery() {
+  return new NgxRewriteOptions(thread_system());
```

Comment:
> can you put in a note

Added comment.  Mostly it points to https://github.com/pagespeed/ngx_pagespeed/issues/1190

> `create_dir_config`

That would be `ps_create_loc_conf`.

But `create_dir_config` isn't relevant here, since we have this bug even without directory-specific options.